frontend/src/components/core/Block/Block.tsx

```diff
@@ -166,27 +142,37 @@ class Block extends PureComponent<Props> {
     width: number
   ): ReactNode | null {
     const element = reportElement.get("element")
-
-    if (element.get("type") === "empty") {
-      // Just a plain div -- so we're sure the rendered element has no
-      // height/margin/padding. Also saves some CPU cycles as a side-effect.
-      return <div className="stEmpty" key={index}></div>
-    }
-
     const component = this.renderElement(
       element,
       index,
       width,
       reportElement.get("metadata")
     )
 
+    const componentWithMaybe = (
+      <Maybe
+        enable={
+          // Note that this will toggle the shouldComponentUpdate flag.
+          // shouldComponentUpdate is used on the update flow, not on the
+          // mount flow. When mounting this check is effectively skipped.
+          element.get("type") !== "empty" ||
+          this.props.reportRunState !== ReportRunState.RUNNING
+        }
+      >
+        {component}
+      </Maybe>
+    )
+
     const isStale =
```

Comment:
i'd move this a separate function

Reply:
I went on a different approach. I moved Maybe to be the outer wrapper. This was needed because I was uncorrectly setting the wrong css on a "Maybed" component which was causing some jumpiness. I can still factor out the Maybe creation, but I feel the Jsx syntaxt is more clear. I can factor out the logic for the condition though. Will put it up, but unsure it is an improvement or it is obfuscating.